client/src/ui/organisms/top-navigation-main/index.tsx

```diff
@@ -20,10 +20,13 @@ export const TopNavigationMain = ({ isOpenOnMobile }) => {
   const [showSearch, setShowSearch] = React.useState(false);
   const [hasOpened, setHasOpened] = React.useState<boolean | undefined>(false);
 
-  function handleShowSearch() {
-    setShowSearch(true);
-    setHasOpened(true);
-  }
+  const handleShowSearch = React.useCallback(function handleShowSearch(
```

Comment:
Can you elaborate why the `function` is wrapped in [`React.useCallback()`](https://reactjs.org/docs/hooks-reference.html#usecallback)?

Reply:
> That's because I need the prop `onChangeIsFocused` and the `onFocus` argument of `useFocusOnSlash` to be referentially stable, that way the `useEffect` won't be triggered on every render. eyes

@davbrito My review comment was about `handleShowSearch`, which used to be a `function`. Does this function need to be referentially stable as well, even though it doesn't have any (read/input) dependencies?